File: ServerHotel-WebManager/ServerHotel-WebManager/models/Room.js
const mongoose = require('mongoose');

const schema = mongoose.Schema;

const roomSchema = new schema({

    
    type: {
        type: String,
        required: true
    },
    price: {
        type: Number,
        require: true
    },
    description: {
        type: String,
        require: false
    },
    convenience: {
        type: Array,
        nameConvenience: {
            type: String,
        },
        imageConvenience: {
            type: String,
        }
    },
    isAvailable: {
        type: Boolean,
        required: true
    },
    image: {
        type: Array,
        required: true
    }
});

const Room = mongoose.model('Room', roomSchema);

module.exports = Room;

File: Hopa-master/Hopa-master/app/src/main/java/com/poly/hopa/models/Room.java
package com.poly.hopa.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Room {
    @SerializedName("_id")
    private String roomId;
    private String type;
    private int price;
    private List<Convenience> convenience;
    private String description;
    private Boolean isAvailable;
    private List<String> image;

    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public List<Convenience> getConvenience() {
        return convenience;
    }

    public void setConvenience(List<Convenience> convenience) {
        this.convenience = convenience;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Boolean getAvailable() {
        return isAvailable;
    }

    public void setAvailable(Boolean available) {
        isAvailable = available;
    }

    public List<String> getImage() {
        return image;
    }

    public void setImage(List<String> image) {
        this.image = image;
    }
}
